community/kernel/src/main/java/org/neo4j/kernel/impl/factory/EditionModule.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.factory;
+
+import org.neo4j.graphdb.DependencyResolver;
+import org.neo4j.kernel.IdGeneratorFactory;
+import org.neo4j.kernel.KernelDiagnostics;
+import org.neo4j.kernel.NeoStoreDataSource;
+import org.neo4j.kernel.impl.api.CommitProcessFactory;
+import org.neo4j.kernel.impl.api.SchemaWriteGuard;
+import org.neo4j.kernel.impl.core.LabelTokenHolder;
+import org.neo4j.kernel.impl.core.PropertyKeyTokenHolder;
+import org.neo4j.kernel.impl.core.RelationshipTypeTokenHolder;
+import org.neo4j.kernel.impl.locking.Locks;
+import org.neo4j.kernel.impl.storemigration.UpgradeConfiguration;
+import org.neo4j.kernel.impl.transaction.TransactionHeaderInformationFactory;
+import org.neo4j.kernel.info.DiagnosticsManager;
+
+/**
+ * Edition module for {@link org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory}. Implementations of this class
+ * need to create all the services that would be specific for a particular edition of the database.
+ */
+public abstract class EditionModule
+{
+    public IdGeneratorFactory idGeneratorFactory;
```

Comment:
There shouldn't be a setter for that - if there is, then something is wrong in the way code is structured.

Having a getter is nice for controlling access and later refactoring.